src/js/angular/import/directives/import-resource-tree.directive.js

```diff
@@ -22,30 +30,93 @@ function importResourceTreeDirective() {
             // Public variables
             // =========================
             $scope.displayResources = [];
+            $scope.TYPE_FILTER_OPTIONS = TYPE_FILTER_OPTIONS;
+            $scope.filterByType = TYPE_FILTER_OPTIONS.ALL;
+            $scope.filterByFileName = '';
 
             // =========================
             // Private variables
             // =========================
             let selectedResources = [];
 
+            // =========================
+            // Public functions
+            // =========================
+            $scope.selectionChanged = (($event, resource) => {
+                resource.setSelection(resource.selected);
+                selectedResources = $scope.resources.getAllSelected();
+            });
+
+            $scope.changeTypeFilter = (newType) => {
+                $scope.filterByType = newType;
+                updateListedImportResources();
+            };
+
+            $scope.handleInputChange = function (filterByFileName) {
```

Comment:
What input change? This should be `filterResources` I guess

Reply:
Renamed.